tests/integrationtest/t/globalindex/mem_index_merge.test

```diff
@@ -0,0 +1,103 @@
+--sorted_result
+select /*+ use_index_merge(tpk2, uidx_a, idx_c) */ * from tpk2 where a > 1 and c > 1;
+--sorted_result
+select /*+ use_index_merge(tpk2, uidx_a, idx_c) */ * from tpk2 where a > 0 and c > 0;
+
+--echo ## for indexMerge union with specified PARTITION
+--replace_regex /in\(_tidb_pid, [0-9]+\)/in(_tidb_pid, pid1)/
+explain select /*+ use_index_merge(tpk2, uidx_a, idx_bc) */ * from tpk2 partition(p1) where a=1 or b=4;
+--sorted_result
+select /*+ use_index_merge(tpk2, uidx_a, idx_bc) */ * from tpk2 partition(p1) where a=1 or b=4;
+
+--echo ## for indexMerge intersection with specified PARTITION
+--replace_regex /in\(_tidb_pid, [0-9]+\)/in(_tidb_pid, pid1)/
+explain select /*+ use_index_merge(tpk2, uidx_a, idx_c) */ * from tpk2 partition(p1) where a > 1 or c > 1;
+--sorted_result
+select /*+ use_index_merge(tpk2, uidx_a, idx_c) */ * from tpk2 partition(p1) where a > 1 or c > 1;
+--sorted_result
+select /*+ use_index_merge(tpk2, uidx_a, idx_c) */ * from tpk2 partition(p1) where a > 0 or c > 0;
+
+rollback;
+
+--echo ## Test CommonHandle
+drop table tpk2;
+CREATE TABLE `tpk2` (
+  `a` int(11) DEFAULT NULL,
+  `b` int(11) DEFAULT NULL,
+  `c` int(11) NOT NULL,
+  `d` int(11) NOT NULL,
+  KEY `idx_bc` (`b`,`c`),
+  UNIQUE KEY `uidx_a` (`a`),
+  UNIQUE KEY `uidx_ac` (`a`, `c`),
+  KEY `idx_c` (`c`),
+  PRIMARY KEY(`d`, `c`)
+) PARTITION BY HASH (`d`) PARTITIONS 5;
+
+insert into tpk2 values (1, 2, 1, 1), (3, 6, 3, 3);
+
+begin;
+insert into tpk2 values (2, 4, 2, 2);
+
+--echo ## for indexMerge union
+explain select /*+ use_index_merge(tpk2, uidx_a, idx_bc) */ * from tpk2 where a=1 or b=4;
+--sorted_result
+select /*+ use_index_merge(tpk2, uidx_a, idx_bc) */ * from tpk2 where a=1 or b=4;
+--sorted_result
+select /*+ use_index_merge(tpk2, uidx_a, idx_bc) */ * from tpk2 where a=2 or b=4;
+
+--echo ## for indexMerge intersection
+explain select /*+ use_index_merge(tpk2, uidx_a, idx_c) */ * from tpk2 where a > 1 and c > 1;
+--sorted_result
+select /*+ use_index_merge(tpk2, uidx_a, idx_c) */ * from tpk2 where a > 1 and c > 1;
+--sorted_result
+select /*+ use_index_merge(tpk2, uidx_a, idx_c) */ * from tpk2 where a > 0 and c > 0;
+
+--echo ## for indexMerge union with specified PARTITION
+--replace_regex /in\(_tidb_pid, [0-9]+\)/in(_tidb_pid, pid1)/
+explain select /*+ use_index_merge(tpk2, uidx_a, idx_bc) */ * from tpk2 partition(p1) where a=1 or b=4;
+--sorted_result
+select /*+ use_index_merge(tpk2, uidx_a, idx_bc) */ * from tpk2 partition(p1) where a=1 or b=4;
+
+--echo ## for indexMerge intersection with specified PARTITION
+--replace_regex /in\(_tidb_pid, [0-9]+\)/in(_tidb_pid, pid1)/
+explain select /*+ use_index_merge(tpk2, uidx_a, idx_c) */ * from tpk2 partition(p1) where a > 1 or c > 1;
+--sorted_result
+select /*+ use_index_merge(tpk2, uidx_a, idx_c) */ * from tpk2 partition(p1) where a > 1 or c > 1;
+--sorted_result
+select /*+ use_index_merge(tpk2, uidx_a, idx_c) */ * from tpk2 partition(p1) where a > 0 or c > 0;
+
+commit;
```

Comment:
Is it intentionally to have the commit here, instead of a rollback in the end? I.e. should the following test not use the mem_reader?

Reply:
Maybe I could execute before and after commit. The plan is a little bit different. In `mem_index_merge`, it not support pushdown order by with limit.